### DISCUSSION TITLE
Back-propagation is too slow with propagate_adjoints

### DISCUSSION THREAD

**Discussion by creotiv** [halide/Halide#5470] [Q&A]
Hi guys, im new to Halide, and have problem with understanding why back-propagation is so slow in comparing with forward run.

Im trying to make operation for pytorch.
Here is my code: https://github.com/creotiv/halide_bilateral_slice_apply

Would be very appreciate if someone can say where is a problem. 
Thanks

**Comment by BachiLi** [comment]
You need to schedule the backward pass. You can use the autoschedulers to do so.

**Reply by creotiv** [reply]
But i used it:

 ```
 output
                    .fuse(x, y, xy)
                    .fuse(xy, c, xyc)
                    .fuse(xyc, n, xycn)
                    .gpu_tile(xycn, tx, 4)
```

**Comment by BachiLi** [comment]
Sorry I missed it. Those are not very efficient schedules. Have you tried the autoschedulers?
